src/commands/appSettings/downloadAppSettings.ts

```diff
@@ -18,7 +18,7 @@ import { getLocalSettingsFile } from "./getLocalSettingsFile";
 
 export async function downloadAppSettings(node?: AppSettingsTreeItem): Promise<void> {
     if (!node) {
-        node = <AppSettingsTreeItem>await ext.tree.showTreeItemPicker(AppSettingsTreeItem.contextValue);
+        node = await ext.tree.showTreeItemPicker<AppSettingsTreeItem>(AppSettingsTreeItem.contextValue);
```

Comment:
Is this just another way to cast?  Why change all the formatting?

Reply:
So that's actually a long story. I changed 
```
public showTreeItemPicker(...): Promise<AzExtTreeItem>;
```

to:
```
public showTreeItemPicker<T extends AzExtTreeItem>(...): Promise<T>;
```

because in Cosmos (the extension I initially used as a trial) typescript was able to infer the return type, which meant I didn't have to cast at all. Basically if `node` is type `AppSettingsTreeItem | undefined` it can infer that `T` should be `AppSettingsTreeItem` _or so I thought_. It turns out that only works if you have `strictNullChecks` off (which I guess Cosmos DB does). Aka it can only infer `T` if the type is exactly `AppSettingsTreeItem` and not `AppSettingsTreeItem | undefined`. So in functions, the inferal doesn't work, and now I have to specify a type for `T` which is why these changed.